lang/py/src/avro/io.py

```diff
@@ -580,16 +696,36 @@ def read_data(self, writers_schema, readers_schema, decoder):
     elif writers_schema.type == 'string':
       return decoder.read_utf8()
     elif writers_schema.type == 'int':
-      return decoder.read_int()
+      if hasattr(writers_schema, 'logical_type'):
+        if writers_schema.logical_type == constants.DATE:
+          return decoder.read_date_from_int()
+        elif writers_schema.logical_type == constants.TIME_MILLIS:
+          return decoder.read_time_millis_from_int()
```

Comment:
I think you need some kind of check to not decode negative ints and longs. The normal int can be between `INT_MIN_VALUE <= INT_MAX_VALUE` and `LONG_MIN_VALUE <= LONG_MAX_VALUE` but I don't suppose you can build a time from negative ints or longs.